src/source/parser.scala

```diff
@@ -26,34 +26,53 @@ import java.util.{Map => JMap}
 import org.yaml.snakeyaml.Yaml
 import scala.collection.JavaConversions._
 import scala.collection.mutable
+import scala.util.control.Breaks._
 import scala.util.parsing.combinator.RegexParsers
 import scala.util.parsing.input.{Position, Positional}
 
-case class Parser() {
+case class Parser(includePaths: List[String]) {
 
 val visitedFiles = mutable.Set[File]()
 val fileStack = mutable.Stack[File]()
 
 private object IdlParser extends RegexParsers {
   override protected val whiteSpace = """[ \t\n\r]+""".r
 
-  def idlFile(origin: String): Parser[IdlFile] = rep(importFile) ~ rep(typeDecl(origin)) ^^ { case imp~types => IdlFile(imp, types) }
-
-  def importFile: Parser[FileRef] = {
-    
-	def fileParent:String = if (fileStack.top.getParent() != null) return fileStack.top.getParent() + "/" else return ""
+  def idlFile(origin: String): Parser[IdlFile] = rep(importFileRef) ~ rep(typeDecl(origin)) ^^ { case imp~types => IdlFile(imp, types) }
 
+  def importFileRef(): Parser[FileRef] = {
     ("@" ~> directive) ~ ("\"" ~> filePath <~ "\"") ^^ {
-      case "import" ~ x =>
-        val newPath = fileParent + x
-        new IdlFileRef(new File(newPath))
-      case "extern" ~ x =>
-        val newPath = fileParent + x
-        new ExternFileRef(new File(newPath))
+      case "import" ~ x => {
+        new IdlFileRef(importFile(x))
+      }
+      case "extern" ~ x => {
+        new ExternFileRef(importFile(x))
+      }
     }
   }
+
+  def importFile(fileName: String): File = {
+    var file: Option[File] = None
+
+    val path = includePaths.find(path => {
+      val relPath = if (path.isEmpty) fileParent else path + "/"
+      val tmp = new File(relPath + fileName)
+      val exists = tmp.exists
+      if (exists) file = Some(tmp)
+      exists
+    })
+
+    if (path.isEmpty || file.isEmpty) throw new FileNotFoundException("Unable to find file \"" + fileName + "\" at " + fileStack.top.getCanonicalPath)
+
+    return file.get
+  }
+
   def filePath = "[^\"]*".r
 
+  def fileParent(): String = {
+    if (fileStack.top.getParent() != null) fileStack.top.getParent() + "/" else ""
```

Comment:
I think it would be better to pass paths around using the File type and use the joining functionality provided by the multi-arg File constructor rather than manipulating separators manually.  It'll be more robust to Windows path separators, among other things.